### DISCUSSION TITLE
[Feature Request]: modifying firmware using gitpod - stable firmware access

### DISCUSSION THREAD

**Discussion by ilker-aktuna** [meshtastic/firmware#4793] [Troubleshooting]
### Platform

ESP32

### Description

I am trying to make a small mod to the firmware using gitpod. 
I can do it on the alpha firmware  using:
[https://gitpod.io#https://github.com/meshtastic/firmware](https://gitpod.io/#https://github.com/meshtastic/firmware)

but I want to do it on the stable firmware (2.4)
So I access using:
[https://gitpod.io#https://github.com/meshtastic/firmware/tree/2.4](https://gitpod.io/#https://github.com/meshtastic/firmware/tree/2.4)

but in this workspace, there is no "pio" to build.
Can it be added ?

**Comment by Talie5in** [comment]
Try with the tags

https://github.com/meshtastic/firmware/tree/v2.4.3.91d6612

**Comment by ilker-aktuna** [comment]
nope:

gitpod /workspace/firmware ((91d6612...)) $ pio
bash: pio: command not found

**Comment by NomDeTom** [comment]
I _think_ if it's older than the gitpod setup files were introduced, you need to do it manually.

Why can't you open it for the latest, and then tell it to fetch the older version?

Edit: yes, you need to run `pip install platformio && pip install --upgrade pip` before you start if the .gitpod.yml isn't in the version you want (i.e. older than about 3 weeks)

**Reply by ilker-aktuna** [reply]
I don't think I have the permission to install anything on the workspace. So "pip install" will not work.
>open it for the latest, and then tell it to fetch the older version?

how ?

**Reply by NomDeTom** [reply]
@ilker-aktuna Have you tried again recently?

**Reply by ilker-aktuna** [reply]
no. I gave up !
